### ISSUE TITLE
Island Reset Crashes

### ISSUE THREAD

**Issue by aulrone** [tastybento/ASkyBlock-Bugs-N-Features#819]
Version: Minecraft 1.12, Spigot most recent, ASkyBlock V3.0.8.8.

Plugins:
BlockParticles, voidchest, EndCraft, PlaceholderAPI, Boss, RandomItems, CommandTimer, SkinsRestorer, DeluxeTags, Java-Version-Tool, ItemID, Minions, DropEdit, ServerHide, ViaVersion, HideStream, ScoreboardLib, ViaBackwards, AsyncWorldEditInjector, CoreProtect, Skript, PowerMining, NBTEditor, ClearLag, BuycraftX, Votifier, KiteBoard, ProtocolLib, NightVision, VotingPlugin, AsyncWorldEdit, Vouchers, VoidGenerator, Aliases, CustomPets, ProPackages, PermissionsEx, TAB, ExperienceBottle, Vault, NameManager, ViaRewind, HolographicDisplays, AuctionHouse, ASkyBlock, EasyRepair, PlayerVaults, CratesPlus, WorldEdit, Essentials, CrateReloaded, VoidSpawn, BannerMaker, Citizens, TradeMe, BankNote, EssentialsChat, NPC_Destinations, ServerShop-GUI, ASkyBlock-IslandBorder, SellStick, WorldGuard, CitizensCMD, LeaderHeads, Oddpicks, WanderingNPC, Eyu-BungeeNPCs, EssentialsSpawn, WorldGuardExtraFlags, BuildersWand, SilkSpawners, Envoys, Skellett, WGRegionEvents, WGCustomFlags, StackMob, WGRegionCommands, ShopGUIPlus

My plugins are legit/up to date.

Issue:

A player starting a new island with /is for the first time has no issue. If they perform /is reset and /is confirm the server gets incredibly laggy and crashes the player/server. This result is the same whether the reset is done on the island or at spawn. The only thing that fixes it is deleting all askyblock player/level files and the world player files.

**Comment by tastybento** [comment]
Is this a new install or an upgrade issue?

What is likely happening is that a clear island spot cannot be found. If you can post the crash log that would be helpful. If that's the case, then what's happening is that the ASkyBlock world generator (that "generates" the void) is not working for some reason and land is being generated everywhere. It's probably another plugin that is interfering somehow although I wouldn't be able to say right now which one.

As we're talking about server crashes here, I recommend you just try ASkyBlock + Vault on a test server. Login, reset, etc. and confirm everything works and then add plugins until something goes wrong. 

Out of all the ones you listed, it's **not** going to be:
PlaceholderAPI, CoreProtect, BuycraftX, Votifier, ProtocolLib,  VotingPlugin, AsyncWorldEdit, PermissionsEx, Vault, HolographicDisplays, ASkyBlock, WorldEdit, Essentials, Citizens, EssentialsChat, WorldGuard, EssentialsSpawn

The rest, I don't know.

**Comment by tastybento** [comment]
@aulrone Any luck?

**Comment by aulrone** [comment]
I found out the issue. The distance between islands was set to 10,000 so whenever a new island is generated it regenerates that ENTIRE 10,000x10,000 radius even if my protection range is only set to 100, which is the actual island area. Thanks for responding, but it would be great if we could space players out more without it breaking the reset feature.

**Comment by Poslovitch** [comment]
10 000 blocks? That's completely pointless, especially when the protection range is only set to 100. With only 9 players, you get a world that could easily be corrupted, and more.
The reset features *needs* to regenerate the whole area : we have had some weird edge-case bugs that made us choose that possibility.

Set the server view-distance down to 4 in Spigot settings and then set the island distance to 200. Players won't be able to see them too much.

@tastybento To avoid such problems in the future, we may need to cap the island distance to a maximum value such as 1000. It'd be completely stupid to have an upper value.

**Comment by aulrone** [comment]
There is a very good reason to have that feature. It's not all about view distances. On a large server, it's crucial for player privacy when they want to only allow tpa's from trusted guests and they don't want the entire server hovering around their base. It makes welcome signs pointless. For popular, top islands I could merely fly 100 blocks in 10 seconds, find out whose island it is, then display the coords. Not everyone will want to box in their entire plot. What about competitive island servers where privacy is important? The askyblock fly feature isn't a togglable or reliable alternative either because it only applies to the island world not the spawn world or anywhere else you would want players to fly.

**Comment by tastybento** [comment]
I understand the need.

The risk with doing protection-only is that there could be some leftover blocks after a reset. For example, if ownership of an island switches from a player with a large protection range to a player with a lower protection range and then they reset. After the reset, the blocks are left and then a new player starts in the old spot, possibly with a larger protection range and suddenly they have a lot of free blocks around the edges. It's not fully exploitable because the players who do the reset won't necessarily get the free blocks, but new random players may.

I've added an option to delete just the protection area and I've actually set it as the default.
